src/emulator/io/src/io.cpp

```diff
@@ -390,333 +288,236 @@ SceOff seek_file(SceUID fd, SceOff offset, int whence, IOState &io, const char *
-        std::string dir_path = to_host_path(translated_path, pref_path, device);
-        fs::remove(dir_path);
-        return 0;
+        statp->st_attr = fd_file->second.get_file_mode();
     }
-    default: {
-        LOG_ERROR("Unimplemented device {} used", dir);
-        return IO_ERROR_UNK();
-    }
-    }
-}
-
-int stat_file(IOState &io, const char *file, SceIoStat *statp, const char *pref_path, uint64_t base_tick, const char *export_name) {
-    assert(statp != NULL);
-
-    memset(statp, '\0', sizeof(SceIoStat));
-
-    std::string translated_path = file;
-    trim_leading_slash(translated_path);
-    VitaIoDevice device = get_device(translated_path);
-
-    // read and execute access rights
-    statp->st_mode = SCE_S_IRUSR | SCE_S_IRGRP | SCE_S_IROTH | SCE_S_IXUSR | SCE_S_IXGRP | SCE_S_IXOTH;
 
     std::uint64_t last_access_time_ticks;
-    std::uint64_t last_modification_time_ticks;
-    std::uint64_t creation_time_ticks{ 0 };
-
-    translate_path(translated_path, device, io.device_paths);
-
-    LOG_TRACE("{}: Statting file {} ({})", export_name, file, translated_path);
-
-    switch (device) {
-    case VitaIoDevice::SA0:
-    case VitaIoDevice::UX0:
-    case VitaIoDevice::UMA0: {
-        std::string file_path = to_host_path(translated_path, pref_path, device);
+    std::uint64_t creation_time_ticks;
 
 #ifdef WIN32
-        WIN32_FIND_DATAW find_data;
-        HANDLE handle = FindFirstFileW(string_utils::utf_to_wide(file_path).c_str(), &find_data);
-        if (handle == INVALID_HANDLE_VALUE) {
-            return IO_ERROR(SCE_ERROR_ERRNO_EMFILE);
-        }
-        FindClose(handle);
-
-        if ((find_data.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) == FILE_ATTRIBUTE_DIRECTORY) {
-            statp->st_mode |= SCE_S_IFDIR;
-        } else {
-            statp->st_mode |= SCE_S_IFREG;
-        }
+    WIN32_FIND_DATAW find_data;
+    const auto handle = FindFirstFileW(file_path.generic_path().wstring().c_str(), &find_data);
+    if (handle == INVALID_HANDLE_VALUE)
+        return IO_ERROR(SCE_ERROR_ERRNO_EMFILE);
 
-        statp->st_size = (std::uint64_t)find_data.nFileSizeHigh << 32 | (std::uint64_t)find_data.nFileSizeLow;
+    FindClose(handle);
 
-        last_access_time_ticks = convert_filetime(find_data.ftLastAccessTime);
-        last_modification_time_ticks = convert_filetime(find_data.ftLastWriteTime);
-        creation_time_ticks = convert_filetime(find_data.ftCreationTime);
+    last_access_time_ticks = convert_filetime(find_data.ftLastAccessTime);
+    creation_time_ticks = convert_filetime(find_data.ftCreationTime);
 #else
-        struct stat sb;
-        if (stat(file_path.c_str(), &sb) < 0) {
-            return IO_ERROR_UNK();
-        }
-
-        if (S_ISREG(sb.st_mode)) {
-            statp->st_mode |= SCE_S_IFREG;
-        } else if (S_ISDIR(sb.st_mode)) {
-            statp->st_mode |= SCE_S_IFDIR;
-        }
-
-        statp->st_size = sb.st_size;
+    struct stat sb;
```

Comment:
Why `struct` in C++? T_T

Reply:
This line is a carry-over from before my PR.

[IEEE specification for getting file status in POSIX](http://pubs.opengroup.org/onlinepubs/009695399/functions/stat.html).

Reply:
That's fair but I'd rather you changed it now that you've found it. T_T